test/new-e2e/tests/agent-platform/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,138 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+package upgrade
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common"
+	filemanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/file-manager"
+	helpers "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/helper"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/install"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/install/installparams"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/platforms"
+	e2eOs "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+	"github.com/stretchr/testify/require"
+	"os"
+	"strings"
+	"testing"
+)
+
+var osVersion = flag.String("osversion", "", "os version to test")
+var platform = flag.String("platform", "", "platform to test")
+var architecture = flag.String("arch", "", "architecture to test (x86_64, arm64))")
+var flavorName = flag.String("flavor", "datadog-agent", "package flavor to install")
+var srcAgentVersion = flag.String("src-agent-version", "5", "start agent version")
+var destAgentVersion = flag.String("dest-agent-version", "7", "destination agent version to upgrade to")
+var cwsSupportedOsVersion = flag.String("cws-supported-osversion", "", "list of os where CWS is supported")
```

Comment:
I think we do not care about CWS in the upgrade tests. You can probably remove everything related to CWS in this test.